packages/sodium/install

```diff
@@ -2,8 +2,27 @@
 set -x
 set -e
 
+OS_DISTRIBUTION=$(lsb_release -cs)
+
 apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C7917B12
-echo "deb http://ppa.launchpad.net/chris-lea/libsodium/ubuntu $(lsb_release -cs) main" >> /etc/apt/sources.list.d/libsodium-ppa.list
+
+function add_repo {
+    echo "deb http://ppa.launchpad.net/chris-lea/libsodium/ubuntu $OS_DISTRIBUTION main" >> /etc/apt/sources.list.d/libsodium-ppa.list
+}
+
+# check for supported platform (whitelisted ubuntu versions)
+if [ $OS_DISTRIBUTION == "precise" ]; then
+    add_repo
+elif [ $OS_DISTRIBUTION == "trusty" ]; then
+    add_repo
+elif [ $OS_DISTRIBUTION == "vivid" ]; then
+    add_repo
+else
+    echo "This platform is unsupported ($OS_DISTRIBUTION) by the apt repository. Skipping"
+    test -f /etc/apt/sources.list.d/libsodium-ppa.list && rm /etc/apt/sources.list.d/libsodium-ppa.list
```

Comment:
I'm torn on this change. It's good that the top-level install script does not err, but it feels like this per-package install script _should_ err.

Thinking about the shinyapps.io case, it would be useful for the user to get some indication that the dependencies for sodium are not available for the distribution being used.

I'd like to get the opinion of the shinyapps folks, where this code is more regularly used. How should the `sodium` dependency installer behave on xenial (which is not supported by its dependencies)? Should we fail here or wait for the R package install to err?

CC @jspiewak @kippandrew @stevenolen

Reply:
Starting with Xenial, libsodium-dev is https://packages.ubuntu.com/xenial/libsodium-dev, and I would suggest simply checking that the Ubuntu version is < Xenial and simplify this change.